cmd/minikube/cmd/tunnel.go

```diff
@@ -116,6 +122,35 @@ var tunnelCmd = &cobra.Command{
 	},
 }
 
+func cleanupLock() {
+	if lockHandle != nil {
+		err := lockHandle.Unlock()
+		if err != nil {
+			out.Styled(style.Warning, fmt.Sprintf("failed to release lock during cleanup: %s", err.Error()))
+		}
+	}
+}
+
+func mustLockOrExit() {
+	tunnelLockPath := filepath.Join(localpath.MiniPath(), ".tunnel_lock")
+
+	_, err := os.OpenFile(tunnelLockPath, os.O_RDWR|os.O_TRUNC, 0600)
+	if err != nil {
+		if !os.IsNotExist(err) {
+			exit.Error(reason.SvcTunnelStart, fmt.Sprintf("error opening lock file for tunnel file (%s)", tunnelLockPath), err)
+		}
+		_, err = os.Create(tunnelLockPath)
+		if err != nil {
+			exit.Error(reason.SvcTunnelStart, fmt.Sprintf("error creating lock for tunnel file (%s)", tunnelLockPath), err)
+		}
+	}
+	lockHandle = fslock.New(tunnelLockPath)
+	err = lockHandle.TryLock()
+	if err == fslock.ErrLocked {
+		exit.Message(reason.SvcTunnelAlreadyRunning, "Another tunnel process is already running, terminate the existing instance to start a new one")
+	}
```

Comment:
We're only checking for `fslock.ErrLocked`, but if the error is not nil but not `fslock.ErrLocked` we should at least `klog.Warningf` it so we can see the cause of the lock error in the logs for debugging pruposes.

Reply:
I updated the code to exit (we should only start the tunnel if we succeed to acquire the lock) if we fail to acquire the lock due to unexpected error, let me know if that is okay!